vra/data_source_image_profile.go

```diff
@@ -0,0 +1,142 @@
+package vra
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/vmware/vra-sdk-go/pkg/client/image_profile"
+	"github.com/vmware/vra-sdk-go/pkg/models"
+
+	"fmt"
+	"log"
+	"strings"
+)
+
+func dataSourceImageProfile() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceImageProfileRead,
+
+		Schema: map[string]*schema.Schema{
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date when the entity was created. The date is in ISO 8601 and UTC.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "A human-friendly description.",
+			},
+			"external_region_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The id of the region for which this profile is defined as in the cloud provider.",
+			},
+			"filter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Filter query string that is supported by vRA multi-cloud IaaS API. Example: regionId eq '<regionId>' and cloudAccountId eq '<cloudAccountId>'.",
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The id of the image profile instance.",
+			},
+			"image_mapping": imageMappingSchema(),
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "A human-friendly name used as an identifier in APIs that support this option.",
+			},
+			"owner": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Email of the user that owns the entity.",
+			},
+			"region_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The if of the region for which this profile is defined as in vRealize Automation(vRA).",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date when the entity was last updated. The date is ISO 8601 and UTC.",
+			},
+		},
+	}
+}
+
+func dataSourceImageProfileRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("Reading the vra_image_profile data source with filter %s", d.Get("filter"))
+
+	apiClient := m.(*Client).apiClient
+	var imageProfile *models.ImageProfile
+	var filter string
+
+	id := d.Get("id").(string)
+	name := d.Get("name").(string)
+	configFilter := d.Get("filter").(string)
+	regionID := d.Get("region_id").(string)
+
+	if id == "" && name == "" && configFilter == "" && regionID == "" {
```

Comment:
Do we need to check that only one of these is set?

Reply:
Done.